lib/galaxy/managers/users.py

```diff
@@ -144,15 +144,13 @@ def create(self, email=None, username=None, password=None, **kwargs):
         else:
             # Activation is off, every new user is active by default.
             user.active = True
-        self.session().add(user)
+        session = self.session()
+        session.add(user)
         try:
-            session = self.session()
-            with transaction(session):
-                session.commit()
-            # TODO:?? flush needed for permissions below? If not, make optional
+            # Creating a private role will commit the session
+            user.attempt_create_private_role()
         except exc.IntegrityError as db_err:
             raise exceptions.Conflict(str(db_err))
-        self.app.security_agent.create_user_role(user, self.app)
```

Comment:
This also sets default permissions, maybe we can alter `security_agent.create_user_role` to bypass
```
        stmt = (
            select(Role)
            .where(
                and_(
                    UserRoleAssociation.user_id == user.id,
                    Role.id == UserRoleAssociation.role_id,
                    Role.type == Role.types.PRIVATE,
                )
            )
            .distinct()
        )
        role = self.sa_session.execute(stmt).scalar_one_or_none()
```
if user.id is None (which means the user is new and not committed to the database, in which case we definitely need to create the user role).

Reply:
Ahh good catch! thanks!